lib/improver/tests/nbhood/use_nbhood/test_CollapseMaskedNeighbourhoodCoordinate.py

```diff
@@ -441,5 +450,36 @@ def test_multidemensional_neighbourhood_input(self):
         result = self.plugin.process(nbhood_cube)
         self.assertArrayAlmostEqual(expected_result, result.data)
 
+    def test_preserve_dimsensions_input(self):
```

Comment:
We should also test that length-one dimensions continue to be dimensions and not scalar coordinates.

Reply:
Done